server/migrations/2024-07-05-221027_update_triggers/up.sql

```diff
@@ -0,0 +1,49 @@
+-- Your SQL goes here
+DROP TRIGGER IF EXISTS update_chunk_metadata_counts_trigger ON chunk_metadata;
+DROP FUNCTION IF EXISTS update_chunk_metadata_counts();
+
+-- Function for INSERT operations
+CREATE OR REPLACE FUNCTION update_chunk_metadata_count_insert() 
+RETURNS TRIGGER AS $$
+BEGIN
+    INSERT INTO dataset_usage_counts (dataset_id, chunk_count)
+    SELECT dataset_id, COUNT(*)
+    FROM inserted
+    GROUP BY dataset_id
+    ON CONFLICT (dataset_id) DO UPDATE
+    SET chunk_count = dataset_usage_counts.chunk_count + EXCLUDED.chunk_count;
```

Comment:
If it's ever the case that there are multiple dataset_id's then they will all get the same count.